sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -130,10 +130,7 @@ object ResolveHints {
           relationNames.foreach(relationNameSet.add)
 
           val applied = applyJoinStrategyHint(h.child, relationNameSet, h.name)
-          relationNameSet.foreach { n =>
-            logWarning(s"Count not find relation '$n' for join strategy hint " +
-              s"'${h.name}${relationNames.mkString("(", ", ", ")")}'.")
-          }
+          hintErrorHandler.handleHintRelationsNotFound(h.name, h.parameters, relationNameSet.toSet)
```

Comment:
This is specific to join hints right? Maybe we should rename `handleHintRelationsNotFound` to something join specific.

Reply:
At this moment we don't have many other hints, but I'd not say it's join specific. Like in other systems, there's index hint that can be and should be bound to a relation.